src/paddle.cpp

```diff
@@ -255,4 +257,123 @@ void Paddle::resetEffects() {
 
 	shield = false;
 	magnetised = false;	
-}
\ No newline at end of file
+}
+
+/*
+ * Paddle AI
+ */
+
+void Paddle::logAction(paddleNS::ACTION a) {
+	char actionStr[10];
+
+	if (a == paddleNS::UP)
+		sprintf(actionStr, "UP");
+	else if (a == paddleNS::DOWN)
+		sprintf(actionStr, "DOWN");
+	else
+		sprintf(actionStr, "STAY");
+
+	printf("[PADDLE AI] %s\n", actionStr);
+}
+
+float Paddle::moveUp() {
+	if (getY() > TOP_WALL) {	// paddle is below the top wall
+		return -paddleNS::VELOCITY * yVelocityMultipler;
+	}
+
+	return 0.0f;
+}
+
+float Paddle::moveDown() {
+	if (getY() + (paddleNS::HEIGHT * getScaleY()) < BOTTOM_WALL) {
+		return paddleNS::VELOCITY * yVelocityMultipler;
+	}
+
+	return 0.0f;
+}
+
+paddleNS::ACTION Paddle::convertNoToAction(int no) {
+	if (no == 0)
+		return paddleNS::UP;
+	if (no == 1)
+		return paddleNS::DOWN;
+	return paddleNS::STAY;
+}
+
+bool Paddle::ballLevelWithPaddle(int centerBallY, int centerPaddleY) {
+	int height = spriteData.height * spriteData.scale.y;
+
+	int max = centerPaddleY + height / 2;
+	int min = centerPaddleY - height / 2 ;
+
+	return centerBallY < max && centerBallY > min;
```

Comment:
If you do it this way, does it mean the paddle will almost always hit the ball using it's edges? And result in many near 90 deg balls.